site/content/entry/citra-progress-report-2023-q2/index.md

```diff
@@ -0,0 +1,504 @@
+
+{{< figure src="download.png"
+    title="A simple way to download all the system files you need for Citra." >}}
+	
+### Implement PrepareToStartApplication, StartApplication, and WakeupApplication ([#6280](https://github.com/citra-emu/citra/pull/6280)) by [Steveice10](https://github.com/Steveice10) and [Subv](https://github.com/Subv/)
+
+With the accessibility issue resolved, Steveice10 could now begin working on making the HOME menu more usable in Citra.
+Continuing on the trend of reviving abandoned projects from the past, this PR implements some important APT service calls used by the HOME Menu in order to start applications. Originally the work of Subv, it addresses one of the longest standing issues with HOME Menu emulation, which was that attempting to launch anything from it would result in an instant crash.
+
+The implementation had been left untouched and neglected for years, until Steveice10 cleaned up the code and got it finally merged into Citra.
+
+The benefits aren’t limited to just the HOME Menu, however. While these functions are mainly used by the HOME Menu, this doesn’t mean that games can’t take advantage of them either. A notable example is Teenage Mutant Ninja Turtles: Master Splinter’s Training Pack, which attempts to shut itself down in order to launch its games, causing an unexpected crash. Now it’s fully playable!
+
+{{< figure src="tmnt.png"
+    title="We're happy to finally see you, Michelangelo!" >}}
+	
+### Fully stub nim:u service. ([#6290](https://github.com/citra-emu/citra/pull/6290)) by [Steveice10](https://github.com/Steveice10)
+
+NIM stands for Network Installation Manager, and is a 3DS system module that handles title installations from the CDN. This in turn creates a bunch of services, of which nim:u is the so-called “updater service”. This is used, for example, for the system updates in the 3DS system settings.
+Without this service properly implemented, Citra would exhibit strange glitches, such as instability around app icons on the HOME Menu, or random downloading of present icons with garbled titles.
+
+Citra did, in fact, have a portion of this module semi-implemented in the past, but it wasn’t exactly put to good use. Back in 2017, Citra developer Subv added the [CheckForSysUpdateEvent syscall](https://github.com/citra-emu/citra/pull/2974), which allowed the HOME Menu to boot.
+However, doing anything other than just looking at the pretty background would softlock the emulator. One of the caveats of this addition was that the LLE NIM applet, used for handling title installations as mentioned above, needed to be enabled from the debugging menu, which wasn’t very user-friendly or obvious.
+
+{{< figure src="lle.png"
+    title="" >}}
+	
+To address this issue, Steveice10 implemented full stubs for the nim:u service based on their own reverse engineering work of the 3DS. 
```

Comment:
```suggestion
To address this issue, Steveice10 implemented stubs for the nim:u service based on their own reverse engineering work of the 3DS. 
```
full stubs is an oxymoron :p

Reply:
“Full” here is meant to mean the whole service has resonable stubs rather than the stubs themselves being “full” in some way, but I agree the word can be left out if it’s confusing.